sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2538,6 +2538,56 @@ case class Encode(value: Expression, charset: Expression)
     newLeft: Expression, newRight: Expression): Encode = copy(value = newLeft, charset = newRight)
 }
 
+/**
+ * Converts the input expression to a binary value based on the supplied format.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(str[, fmt]) - Converts the input `str` to a binary value based on the supplied `fmt`.
+      By default, the binary format for conversion is "hex" if `fmt` is omitted.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('abc', 'utf-8');
+       abc
+  """,
+  since = "3.3.0",
+  group = "string_funcs")
+// scalastyle:on line.size.limit
+case class ToBinary(expr: Expression, format: Option[Expression], child: Expression)
+  extends RuntimeReplaceable {
+
+  def this(expr: Expression, format: Expression) = this(expr, Option(format),
+    format.toString match {
+      case "hex" => Unhex(expr)
+      case "utf-8" => Encode(expr, Literal("UTF-8"))
+      case "base64" => UnBase64(expr)
+      case "base2" => Cast(expr, BinaryType)
+    }
+  )
```

Comment:
We would have to do this like:

```scala
  def this(expr: Expression, format: Expression) = this(expr, Option(format),
    format match {
      case Literal("hex", StringType) => Unhex(expr)
      case Literal("utf-8", StringType) => Encode(expr, Literal("UTF-8"))
      case Literal("base64", StringType) => UnBase64(expr)
      case Literal("base2", StringType) => Cast(expr, BinaryType)
      case other => other
    }
  )
```

So later this argument can be checked `checkInputDataTypes` instead of failing in the constructor.

Reply:
The code block works perfectly. Thanks!

Is `format` something like `Literal("hex")`? 
`Literal("hex")` doesn't seem to match `Literal("hex", StringType)` as below:
```
scala> val format = Literal("hex")
format: org.apache.spark.sql.catalyst.expressions.Literal = hex

scala> format match {
     |   case Literal("hex", StringType) => println("matched")
     |   case _ => println("not matched")
     | }
not matched
```

Reply:
Yeah, `Literal("hex")` creates a literal with internal data. `Literal(UTFString("hex"), StringType)`. Actually, this is a good point. I think we should do this something like:

```scala
format match {
  case lit: Literal if lit.dataType == StringType => lit.eval() match {
    case "hex" => ...
  }
  case other => ...
}
```

or

```scala
format match {
  case lit if lit.foldable => lit.eval() match {
    case "hex" => ...
  }
  case other => ...
}
```